ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -334,4 +333,17 @@ private void checkStarted()
       throw new StoreException("Store not started", StoreErrorCodes.Store_Not_Started);
     }
   }
+
+  /**
+   * Sets the end offsets and states for all the segments (by setting the active segment).
+   * @throws IOException if there is an I/O error while setting the end offsets.
+   */
+  private void setSegmentStatesAndEndOffsets()
```

Comment:
Yes that's correct - the presence of this function is (also was) extraneous because the index already does this. Depending how the index evolves, I might get rid of this function.